lib/Doctrine/ORM/AbstractQuery.php

```diff
@@ -1044,7 +1046,7 @@ protected function getHydrationCacheId(): array
         ksort($hints);
         assert($queryCacheProfile !== null);
 
-        return $queryCacheProfile->generateCacheKeys($sql, $parameters, $hints);
+        return $queryCacheProfile->generateCacheKeys($sql, $parameters, $types, $hints);
```

Comment:
This is actually an interesting find. We've called this method the wrong way the whole time. 🫣

Reply:
Backported as #10355.